### PR TITLE
Feature/fe add proposal

### PR DESCRIPTION
**Pull request by leinss**

### CODE REVIEW

File: Frontend/src/pages/AddProposal.tsx [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+        </FormHelperText>
+      </FormControl>
+
+      <Button
+        colorScheme="teal"
+        isDisabled={isSubmitting}
+        onClick={() => onSubmit()}
+      >
+        {isSubmitting ? <Spinner /> : "Submit"}
+      </Button>
```
Comment (Bitcoinera) [inline Frontend/src/pages/AddProposal.tsx]: I think this button should have more padding and look more compact, more like a candy.
This is how it's looking on my laptop:

image

Comment (Bitcoinera) [reply Frontend/src/pages/AddProposal.tsx]: Also, this button should be disabled until we have the addProposal function ready on the smart contract

**Comment by Bitcoinera** [comment]
I see this PR also includes changes from #29

File: Frontend/src/pages/AddProposal.tsx [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    }
+    if (!input.ipfsFolderCID) {
+      setIsError({ ...isError, ipfsFolderCID: true })
+      return
+    }
+    setIsError({ collectionSize: false, ipfsFolderCID: false })
+
+    setIsSubmitting(true)
+    await daoContract.addProposal(
+      input.collectionSize,
```
Comment (Bitcoinera) [inline Frontend/src/pages/AddProposal.tsx:54]: What do we need `collectionSize` for, actually?

Comment (leinss) [reply Frontend/src/pages/AddProposal.tsx:54]: In the Ballot
  
  struct Proposal {
    uint256 index; // index for the NFT collection
    uint256 voteCount; // number of accumulated votes
    uint256 collectionSize; // number of items in collection starting from 0
    bool active; // the proposal is still being voted on
    string ipfsFolderCID;
  }

**Comment by leinss** [comment]
> I see this PR also includes changes from #29

Yes, I need to work with the changes of #29 that why I want to merge.

File: Frontend/src/pages/AddProposal.tsx [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+  Heading,
+  Input,
+  NumberDecrementStepper,
+  NumberIncrementStepper,
+  NumberInput,
+  NumberInputField,
+  NumberInputStepper,
+  Spinner,
+} from "@chakra-ui/react"
+import { useContext, useState } from "react"
```
Comment (Bitcoinera) [inline Frontend/src/pages/AddProposal.tsx:16]:
```suggestion
import { useContext, useState } from "react"
import { useSigner } from "wagmi"
```

File: Frontend/src/pages/AddProposal.tsx [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+    ipfsFolderCID: "",
+  })
+  const [isError, setIsError] = useState({
+    collectionSize: false,
+    ipfsFolderCID: false,
+  })
+
+  const [isSubmitting, setIsSubmitting] = useState<boolean>(false)
+
+  const { daoContract } = useContext(StateContext)
```
Comment (Bitcoinera) [inline Frontend/src/pages/AddProposal.tsx:35]:
```suggestion
  const { daoContract } = useContext(StateContext)

  const { data: signer } = useSigner()
```

File: Frontend/src/pages/AddProposal.tsx [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+        colorScheme="teal"
+        // remove '|| true' after backend implementation is set
+        isDisabled={isSubmitting || true}
+        onClick={() => onSubmit()}
+        w="fit-content"
+      >
+        {isSubmitting ? <Spinner /> : "Submit"}
+      </Button>
+    </ContentWrapper>
+  )
```
Comment (Bitcoinera) [inline Frontend/src/pages/AddProposal.tsx]:
```suggestion
  return signer ? (
    <ContentWrapper>
      <Heading>Add Proposal</Heading>

      <FormControl mt={10}>
        <FormLabel>Collection Size</FormLabel>

        <NumberInput
          min={0}
          step={1}
          value={input.collectionSize}
          onChange={(value) =>
            setInput({
              ...input,
              collectionSize: value,
            })
          }
          isInvalid={isError.collectionSize}
        >
          <NumberInputField />
          <NumberInputStepper>
            <NumberIncrementStepper />
            <NumberDecrementStepper />
          </NumberInputStepper>
        </NumberInput>
        <FormHelperText>
          How many items does your collection have?
        </FormHelperText>

        <FormLabel mt={5}>IPFS Folder CID</FormLabel>
        <Input
          type="text"
          value={input.ipfsFolderCID}
          onChange={(e) =>
            setInput({
              ...input,
              ipfsFolderCID: e.target.value,
            })
          }
          isInvalid={isError.ipfsFolderCID}
        />
        <FormHelperText>
          Please label your collection 0.json, 1.json, ... inside the folder
        </FormHelperText>
      </FormControl>

      <Button
        colorScheme="teal"
        // remove '|| true' after backend implementation is set
        isDisabled={isSubmitting || true}
        onClick={() => onSubmit()}
        w="fit-content"
      >
        {isSubmitting ? <Spinner /> : "Submit"}
      </Button>
    </ContentWrapper>
  ) : (
    <div>Login with your wallet</div>
  )
```

File: Frontend/src/pages/AddProposal.tsx (hunk 3 above)
Comment (Bitcoinera) [reply Frontend/src/pages/AddProposal.tsx:16]: This is to add account is logged in verification

**Review by Bitcoinera — APPROVED** [review]

**Pull request merged**